agi_check_ili_export/build.gradle

```diff
@@ -0,0 +1,88 @@
+
+description = """\n
+Exportiert sämtliche modellbasierten Datenerfassungen in eine 
+INTERLIS-Transferdatei, um die Modellkonformität zu prüfen.
+
+In Zukunft könnte auch eine direktere Prüfung in der Datenbank
+stattfinden, ohne zu versuchen die Datei zu exportieren
+(Feature request für ilivalidator).
+"""
+
+apply plugin: 'ch.so.agi.gretl'
+
+defaultTasks 'checkAllModels'
+
+ext {
+    todaysDate = new Date().format('yyyy-MM-dd')
+    models = []
+    models.add(["SO_Agglomerationsprogramme_20170512", "arp_aggloprogramme", "arp_aggloprogramme_" + todaysDate ])
+    models.add(["SO_ARP_Nutzungsvereinbarung_20170512", "arp_nutzungsvereinbarung", "arp_nutzungsvereinbarung_" + todaysDate ])
+    models.add(["SO_Forstreviere_20170512", "awjf_forstreviere", "awjf_forstreviere_" + todaysDate ])
+    models.add(["SO_Hoheitsgrenzen_20170623", "agi_hoheitsgrenzen", "agi_hoheitsgrenzen_" + todaysDate ])
+    models.add(["SO_AWJF_Wegsanierungen_20170629", "awjf_wegsanierungen", "awjf_wegsanierungen_" + todaysDate ])
+}
+
+def pathToTempFolder = System.getProperty("java.io.tmpdir")
+
+// Create a dynamic task for every model we want to validate and for cleaning up if validaton was successful.
+models.each { model ->
+    def modelName = model.getAt(0)
+    def dbSchema = model.getAt(1)
+    def dataExists
+
+    
+    task "checkModel_$modelName"(type: Ili2pgExport) {
+        description = "INTERLIS validation against database schema: $dbSchema ($modelName)"
+        database = [dbUriSogis, dbUserSogis, dbPwdSogis]
+        dbschema = dbSchema
+        models = modelName
+        disableValidation = false
+        logFile = file(Paths.get(pathToTempFolder.toString(), dbSchema+".log"))
+        dataFile = file(Paths.get(pathToTempFolder.toString(), dbSchema+".xtf"))
+
+        finalizedBy "removeFiles_$modelName"
+    }
+
+    task "checkDataFileForData_$modelName"(){
+        description = "Temporary workaround until ili2pgExport works correctly on failure during export. " +
+                "Checks if there is any data within datasection."
+        doLast {
+            def dataFile = file(Paths.get(pathToTempFolder.toString(), dbSchema + ".xtf"))
+            def indexStartDatasection = dataFile.findLastIndexOf { it =~ "<DATASECTION>" }
+            def indexEndDatasection = dataFile.findLastIndexOf { it =~ "</DATASECTION>" }
+            def lineDifference= indexEndDatasection - indexStartDatasection
+
+            if (lineDifference <= 1){
+                dataExists = false
+                throw new GradleException("No Data exported")
+            } else if(lineDifference > 1) {
+                dataExists = true
+            }
+        }
+    }
+
+    task "removeFiles_$modelName"(type: Delete, dependsOn:"checkDataFileForData_$modelName") {
+        description = "Remove files from model export: $modelName"
+
+        onlyIf {
+            //project.getTasksByName("checkModel_$modelName", false).getAt(0).state.failure == null
```

Comment:
Wozu genau diese auskommentierte Zeile?

Reply:
Diese Zeile prüft, ob der Task checkModel_modellname erfolgreich durchgelaufen ist. Zur Zeit läuft der Task auch dann erfolgreich durch, wenn er gar nichts exportieren kann (aufgrund von Fehlern in den Daten). Sobald ili2pg-Task korrigiert/angepasst ist, kann dieses Zeile wieder aktiviert werden und der Task checkDataFileForData_modellname entfernt werden.